qsr_lib/src/qsrlib_qsrs/qsr_rcc_abstractclass.py

```diff
@@ -0,0 +1,174 @@
+        # Cond1. If A's left edge is to the right of the B's right edge, - then A is Totally to right Of B
+        # Cond2. If A's right edge is to the left of the B's left edge, - then A is Totally to left Of B
+        # Cond3. If A's top edge is below B's bottom edge, - then A is Totally below B
+        # Cond4. If A's bottom edge is above B's top edge, - then A is Totally above B
+        
+        #    Cond1           Cond2          Cond3         Cond4
+        if (ax-q > dx+q) or (bx+q < cx-q) or (ay-q > dy+q) or (by+q < cy-q):
+            return "dc"
+
+        # Is one object inside the other ()
+        BinsideA = (ax <= cx) and (ay <= cy) and (bx >= dx) and (by >= dy)
+        AinsideB = (ax >= cx) and (ay >= cy) and (bx <= dx) and (by <= dy)
+
+        # Do objects share an X or Y (but are not necessarily touching)
+        sameX = (abs(ax - cx)<=q) or (abs(ax - dx)<=q) or (abs(bx - cx)<=q) or (abs(bx - dx)<=q)
+        sameY = (abs(ay - cy)<=q) or (abs(ay - dy)<=q) or (abs(by - cy)<=q) or (abs(by - dy)<=q)
+        
+        if AinsideB and (sameX or sameY):
+            return "tpp"
+
+        if BinsideA and (sameX or sameY):
+            return "tppi"
+
+        if AinsideB:
+            return "ntpp"
+
+        if BinsideA:
+            return "ntppi"
+
+        similarX = (abs(ax - cx)<q) or (abs(ax - dx)<q) or (abs(bx - cx)<q) or (abs(bx - dx)<q)
+        similarY = (abs(ay - cy)<q) or (abs(ay - dy)<q) or (abs(by - cy)<q) or (abs(by - dy)<q)
+
+        # Are objects touching?
+        # Cond1. If A's left edge is equal to B's right edge, - then A is to the right of B and touching
+        # Cond2. If A's right edge is qual to B's left edge, - then A is to the left of B and touching
+        # Cond3. If A's top edge equal to B's bottom edge, - then A is below B and touching
+        # Cond4. If A's bottom edge equal to B's top edge, - then A is above B and touching
+
+        # If quantisation overlaps, but bounding boxes do not then edge connected,
+        # include the objects edges, but do not include the quantisation edge
+        if ((cx-q) <= (bx+q)) and ((cx-q) >= (bx)) or \
+           ((dx+q) >= (ax-q)) and ((dx+q) <= (ax)) or \
+           ((cy-q) <= (by+q)) and ((cy-q) >= (by)) or \
+           ((dy+q) >= (ay-q)) and ((dy+q) <= (ay)):
+            return "ec"
+
+        # If none of the other conditions are met, the objects must be parially overlapping
+        return "po"
+
+    @abstractmethod
+    def convert_to_current_rcc(self, qsr):
+        """Overwrite this function to filter and return only the relations
+        coresponding to the particular RCC version you are using.
+        
+        Example for RCC2: return qsr if qsr =="dc" else "c"
+
+        :param qtc: The RCC8 relation between two objects
+        :return: The part of the tuple you would to have as a result
+        """
+        raise NotImplementedError("Filter RCC results using convert_to_current_rcc function")
```

Comment:
Isn't that the point of abstract methods? If you don't implement them then you shouldn't be able to instantiate that class. To be fair, I don't think you need this Exception because the `@abstractmethod` decorator will cause an exception as soon as you try to instantiate a child class that does not implement `convert_to_current_rcc` but it shouldn't hurt either.